packages/trpc/server/routers/loggedInViewer/checkForInvalidAppCredentials.ts

```diff
@@ -0,0 +1,38 @@
+import { getAppFromSlug } from "@calcom/app-store/utils";
+import { type InvalidAppCredentialBannerProps } from "@calcom/features/users/components/InvalidAppCredentialsBanner";
+import { prisma } from "@calcom/prisma";
+import type { TrpcSessionUser } from "@calcom/trpc/server/trpc";
+
+type checkInvalidAppCredentialsOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+};
+
+export const checkInvalidAppCredentials = async ({ ctx }: checkInvalidAppCredentialsOptions) => {
+  const apps = await prisma.credential.findMany({
+    where: {
+      userId: ctx.user.id,
+      invalid: true,
+    },
+    select: {
+      app: {
+        select: {
+          slug: true,
+        },
+      },
+    },
+  });
+
```

Comment:
we might also wanna check team credentials that the user is admin/owner of

Reply:
ok